pkg/codegen/nodejs/gen_program.go

```diff
@@ -415,7 +423,7 @@ func (g *generator) collectProgramImports(program *pcl.Program) programImports {
 			if call, ok := n.(*model.FunctionCallExpression); ok {
 				if i := g.getFunctionImports(call); len(i) > 0 && i[0] != "" {
 					for _, importPackage := range i {
-						importSet.Add(importPackage)
+						importSet0.Add(importPackage)
```

Comment:
This is really where the problem was. `g.getFunctionImports` is ignoring `def.Language["nodejs"].(NodePackageInfo)`. Unlike `*pcl.Resource` nodes, `*model.FunctionCallExpression` does not have a handy `.Schema.PackageReference` I could find. If that is available in PCL that'd be much cleaner that the way the code is written now.